### PR TITLE
feat(upshot): Update assetAddress param

### PR DESCRIPTION
**Pull request by 1marcghannam**

### CODE REVIEW

**Review by ericjaurena — CHANGES_REQUESTED** [review]
in README.md lines 27 and 85 should be changed to reflect that the prices returned by the Upshot adapter are measured in `WEI` and not `GWEI`

File: packages/sources/upshot/package.json [hunk 1]
```diff
@@ -30,6 +30,8 @@
   "dependencies": {
```
Comment (vnavascues) [inline packages/sources/upshot/package.json:30]: You must bump the version to `1.1.0` after adding a new endpoint

File: packages/sources/upshot/src/endpoint/statistics.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+export const supportedEndpoints = ['statistics']
+
+export const endpointResultPaths = {}
+
+const customError = (data: ResponseSchema) => {
+  return data.status === false
+}
+
+export const description =
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/statistics.ts:15]: This is outdated. Fix it and regenerate the docs

File: packages/sources/upshot/src/endpoint/asset-price.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+export const supportedEndpoints = ['asset-price']
+
+export const endpointResultPaths = {}
+
+const customError = (data: ResponseSchema) => {
+  return data.status === false
+}
+
+export const description =
+  'This endpoint allows you to query the price in WEI for a specific NFT asset.'
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/asset-price.ts]: `This endpoint allows you to query the floor price in wei for a specific NFT asset.`

`wei` is not in capital case, `wei` or `Wei` (after a period). ETH although is accepted!

By making the price returned is the "floor" then you can simplify `assetAddress` and `tokenId` descriptions:
- `The NFT collection address`
- `The NFT ID`

File: packages/sources/upshot/src/endpoint/asset-price.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const options = { ...config.api, params, url }
+
+  const response = await Requester.request<ResponseSchema>(options, customError)
+
+  let estimatedPrice: number
+
+  try {
+    estimatedPrice = Number(response.data.data[0].currentPricing.estimatedPrice)
+  } catch (error) {
+    const message = (error as Error).message
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/asset-price.ts]: This is not needed, check the `Error` documentation.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message

`${error}` will call `toString()` which already includes `message`.

File: packages/sources/upshot/src/endpoint/asset-price.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    throw new AdapterError({
+      jobRunID,
+      statusCode: 200,
+      message: `Unexpected error occurred parsing the estimated price from the API response. ${message}`,
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/asset-price.ts]: `blabla message. ${error}`

File: packages/sources/upshot/src/endpoint/statistics.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  const options = { ...config.api, params, url }
+
+  const response = await Requester.request<ResponseSchema>(options, customError)
+
+  let result
+
+  try {
+    result = parseResult(response.data.data.stats[0])
+  } catch (error) {
+    const message = (error as Error).message
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/statistics.ts]: This is not needed, check the `Error` documentation.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message

`${error}` will call t`oString()` which already includes `message`.

File: packages/sources/upshot/src/endpoint/statistics.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    throw new AdapterError({
+      jobRunID,
+      statusCode: 200,
+      message: `Unexpected error occurred forming the result from the API response. ${message}`,
+      cause: error,
+      url: join(options.baseURL, options.url),
+    })
+  }
+
+  response.data.result = result
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/statistics.ts:57]: Maaaaaaarc, result is a String, you should return `withResult`

https://github.com/linkpoolio/external-adapters-js/pull/132#discussion_r863663006

File: packages/sources/upshot/src/endpoint/statistics.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  },
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+
+  const jobRunID = validator.validated.id
+  const assetAddress = validator.validated.data.assetAddress
+
+  const url = `/v1/collections/contractaddress/${assetAddress}`
```
Comment (vnavascues) [inline packages/sources/upshot/src/endpoint/statistics.ts]: you should use `buildUrlPath`

**Review by vnavascues — APPROVED** [review]
Excellent! approved it. Just take care of the minor changes before merging it :)

File: packages/sources/upshot/src/endpoint/statistics.ts (hunk 7 above)
Comment (vnavascues) [reply packages/sources/upshot/src/endpoint/statistics.ts:57]: Apologies, it is indeed an object. You did right!

**Pull request merged**